tools/sqliteliboptions.py

```diff
@@ -1,13 +1,23 @@
 #!/usr/bin/env python3
 
-import sys
-import os
+import argparse
 import ctypes
+import ctypes.util
 
-if len(sys.argv) != 2 or not os.path.isfile(sys.argv[1]):
-    sys.exit("Expected filename of sqlite3 shared library/dll")
 
-lib = ctypes.cdll.LoadLibrary(sys.argv[1])
+argp = argparse.ArgumentParser()
+argp.add_argument("library",
+                  help="Path or filename to sqlite3 shared library/dll "
+                       "(as accepted by ctypes.cdll.LoadLibrary(), default: "
+                       "use find_library())",
+                  nargs="?")
+args = argp.parse_args()
```

Comment:
I suspect you may think using `argparse` in a trivial tool is an overkill but I don't think it really adds much extra code here and the help message is much more helpful.

Reply:
The intention was for the code to end up in setup.py which has its own arg parsing setup.  Specifically it requires using inheritance, 3 methods and 1 tuple listing options.  Hence the quick hack.